Snowflake.Data/Core/Session/ConnectionPoolManager.cs

```diff
@@ -85,39 +91,39 @@ public long GetTimeout()
         {
             s_logger.Debug("ConnectionPoolManager::GetTimeout");
             var values = _pools.Values.Select(it => it.GetTimeout()).Distinct().ToList();
-            return values.Count == 1
-                ? values.First()
-                : throw new SnowflakeDbException(SFError.INCONSISTENT_RESULT_ERROR, "Multiple pools have different Timeout values");
+            switch (values.Count)
+            {
+                case 0:
+                    return (long) SFSessionHttpClientProperties.DefaultExpirationTimeout.TotalSeconds;
+                case 1:
+                    return values.First();
+                default:
+                    throw new SnowflakeDbException(SFError.INCONSISTENT_RESULT_ERROR, "Multiple pools have different Timeout values");
+            }
         }
 
         public int GetCurrentPoolSize()
         {
             s_logger.Debug("ConnectionPoolManager::GetCurrentPoolSize");
-            var values = _pools.Values.Select(it => it.GetCurrentPoolSize()).Distinct().ToList();
-            return values.Count == 1
-                ? values.First()
-                : throw new SnowflakeDbException(SFError.INCONSISTENT_RESULT_ERROR, "Multiple pools have different Current Pool Size values");
+            return _pools.Values.Select(it => it.GetCurrentPoolSize()).Sum();
         }
 
         public bool SetPooling(bool poolingEnabled)
         {
             throw s_operationNotAvailable;
         }
 
-        public bool GetPooling() 
+        public bool GetPooling()
         {
             s_logger.Debug("ConnectionPoolManager::GetPooling");
-            var values = _pools.Values.Select(it => it.GetPooling()).Distinct().ToList();
-            return values.Count == 1
-                ? values.First()
-                : throw new SnowflakeDbException(SFError.INCONSISTENT_RESULT_ERROR, "Multiple pools have different Pooling values");
+            return true; // in new pool pooling is always enabled by default, disabling only by connection string parameter
```

Comment:
but this method is a proxy to an old pool so it will always say true even when a user disables pooling on a cache pool

Reply:
no this class ConnectionPoolManager is for new pool only. For old pool it is a class ConnectionCacheManager.
If the user disables pooling on old pool he will be using ConnectionCacheManager class.